contracts/token/oracles/OracleAggregator.sol

```diff
@@ -27,8 +28,9 @@ abstract contract OracleAggregator is Ownable, IOracleAggregator {
         if (decimals1 != decimals2) revert MismatchInBaseAndQuoteDecimals();
         tokensInfo[token].tokenOracle = tokenOracle;
         tokensInfo[token].nativeOracle = nativeOracle;
-        tokensInfo[token].tokenDecimals = tokenDecimals;
+        tokensInfo[token].tokenDecimals = ERC20(token).decimals();
         tokensInfo[token].isDerivedFeed = isDerivedFeed;
+        tokensInfo[token].priceUpdateThreshold = priceUpdateThreshold;
```

Comment:
Current function `setTokenOracle` updates TokenInfo in tokensInfo map separately. 

I'll suggest to pass TokenInfo directly and update map once. Saves SSTORE operation
```
tokensInfo[token] = TokenInfo({
    tokenDecimals: ERC20(token).decimals(),
    priceUpdateThreshold,
    tokenOracle,
    nativeOracle,
    isDerivedFeed
});
```
This way we use less gas

Reply:
can we do this as part of another PR? first I will check gas consumptions on forge. this also requires major changes again in scripts